x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/sort/BytesRefBucketedSortTests.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data.sort;
+
+import org.apache.lucene.document.InetAddressPoint;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.BytesRefBlock;
+import org.elasticsearch.compute.data.ElementType;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.search.sort.SortOrder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class BytesRefBucketedSortTests extends BucketedSortTestCase<BytesRefBucketedSort, BytesRef> {
+    @Override
+    protected BytesRefBucketedSort build(SortOrder sortOrder, int bucketSize) {
+        return new BytesRefBucketedSort(new NoopCircuitBreaker("test"), "test", bigArrays(), sortOrder, bucketSize);
```

Comment:
I wonder if we could replace that Noop with a CircuitBreaker provided by this (PR ready): https://github.com/elastic/elasticsearch/pull/113105/files#diff-4775cf1e524e3d7085511b07a0119726841f6e4566a66cc317617004e7a1cb1bR581-R594

So we also make sure the BucketedSort doesn't leak anything.

Reply:
I think there's a few neat things we could do with these tests:
1. CrankyCircuitBreaker
2. A version that always throws - there's something in OperatorTestCase that does it

We already have the coverage for these in the tests for the aggregators, but I think it'd help to have them here.

FWIW, I'm going to replace this line with on the gets the REQUEST breaker from the `bigArrays`